feat: updated to fetch 0.4.2 binary

wadeking98 opened the pull request:
Updated the javascript wrappers to use the 0.4.2 binary

genaris commented on wrappers/javascript/indy-vdr-nodejs/README.md:
```diff
@@ -51,3 +51,13 @@ yarn test:local-build
+| ------------- | ------------------ |
+| v0.4.0-dev.16 | v0.1.0             |
+| v0.4.1        | v0.2.0             |
+| v0.4.2        | v0.2.3             |
```
I think it will be clearer if we explicitly mention that v0.4.1 corresponds to 0.2.0, 0.2.1 and 0.2.2, or just 0.2.x and call this one (the one under development) as 0.3.0.

Actually in any case I'd prefer to move on to 0.3.0

genaris reviewed (commented):
I think it is good to put the efforts in the main branch but we just need to make sure that it works properly in both nodejs and react-native, as mentioned in https://github.com/hyperledger/indy-vdr/issues/268#issuecomment-2075643725. I didn't follow the latest releases (other than some minor fixes) but I guess the `js-0.4.1` branch was created for a compatibility issue so it would be nice to get feedback from @TimoGlastra or @berendsliedrecht.

Besides that, I remember there were several methods not implemented in the wrappers even for 0.4.1, so it would be good also to take a look if they are already there now or they need to be added before doing another official release: see https://github.com/hyperledger/indy-vdr/issues/216#issuecomment-1723649279

swcurran commented:
I’d love to get this into a single branch and to have unified releases of all components at the same time. Additional overhead, but pretty important!

wadeking98 commented:
@genaris is there an easy way to test it on node and react native? from my understanding it only pulls the binaries when you run npm install from the npm repository.

TimoGlastra commented on wrappers/javascript/indy-vdr-react-native/cpp/turboModuleUtility.cpp:
```diff
@@ -143,7 +143,7 @@ int64_t jsiToValue(jsi::Runtime &rt, jsi::Object &options, const char *name,
   jsi::Value value = options.getProperty(rt, name);
   if ((value.isNull() || value.isUndefined()) && optional)
-    return 0;
+    return -1;
```
Is it correct to return -1 always in this case? 

In that case the above code that transforms it to -1 if it's `0` can also be removed

berendsliedrecht replied:
Yes, after a quick check again -1 is interpreted as `None`. 

@wadeking98 could you fix all the `0`s to `-1`?

wadeking98 replied:
@berendsliedrecht I've done it for int64 and int32 but I'm not sure what to do for uint 64,32,8 because it needs to return unsigned

TimoGlastra commented on wrappers/javascript/indy-vdr-react-native/native/indy_vdr.node:
```diff
@@ -1,3 +0,0 @@
-# This is a placeholder file to prevent node-pre-gyp from installing the indy_vdr binary when cloning
```
This needs to stay!

wadeking98 replied:
sorry about that, I was testing some things in nodejs and didn't mean to commit it

TimoGlastra commented:
@berendsliedrecht maybe we can also add the example app to this branch so we can more easily test changes in react native?

TimoGlastra commented:
> I’d love to get this into a single branch and to have unified releases of all components at the same time. Additional overhead, but pretty important!

I think that makes sense, however it means we will need to require every pull request to implement functionality for all wrappers. It's sometimes quite hard to look in which way a wrapper is outdated, as you have to walk through all commits, see the changes, update it, test the different wrappers, etc..

wadeking98 commented:
I made some changes to the binary to make it work with react native. For some reason the `#[async_trait]` macro wasn't working well with react native so I removed it and refactored the code to compensate. I'm going to split up this PR to update the binary and one to update the wrapper

TimoGlastra commented:
I think this PR can be closed right @wadeking98? I made #296 with updates and a new version for the JS wrapper
